src/widgets/WidgetApi.ts

```diff
@@ -98,6 +108,16 @@ export class WidgetApi {
 
                     // Automatically acknowledge so we can move on
                     this.replyToRequest(<ToWidgetRequest>payload, {});
+                } else if (payload.action === KnownWidgetActions.Terminate) {
+                    // Finalization needs to be async, so postpone with a promise
+                    let finalizePromise = Promise.resolve();
+                    const wait = promise => {
```

Comment:
We generally try to wrap arguments with redundant parens when using const-defined functions:
```suggestion
                    const wait = (promise) => {
```